### ISSUE TITLE
Integration test failure: openTextDocument, untitled closes on save

### ISSUE THREAD

**Issue by alexr00** [microsoft/vscode#157897]
Failed in https://dev.azure.com/vscode/VSCode/_build/results?buildId=70964&view=logs&j=a8a6689c-5a84-59c0-482a-7dd9f21eb2d8&t=1fa6fb8d-7bb3-5acd-5778-bf99aac7f637

```
MainThreadFileSystem#$watch(): failed to stat a resource for file watching (extension: vscode.vscode-api-tests, path: watcherTest:/somePath/folder, recursive: true, session: 0.4519563871799428): EntryNotFound (FileSystemError): watcherTest:/somePath/folder
    ✔ createFileSystemWatcher
  206 passing (16s)
  63 pending
  1 failing
  1) vscode API - workspace
       openTextDocument, untitled closes on save:

      AssertionError [ERR_ASSERTION] [ERR_ASSERTION]: The expression evaluated to a falsy value:

  assert.ok(closed === doc)

      + expected - actual

      -false
      +true

```

**Comment by bpasero** [comment]
Again, could be a regression from https://github.com/microsoft/vscode/commit/dced70bbf36d3c53c08e791da1791ac7fc42519b, we should probably check statistics for previous failures.

One actual change from https://github.com/microsoft/vscode/commit/dced70bbf36d3c53c08e791da1791ac7fc42519b is that `openEditor` will open the editor sync anymore, but always async, so this could have an impact on code that relies on ordering.

In other words:
- old: `group.openEditor` was reached without yielding in the same frame
- new: `group.openEditor` is now reached after an `async await` at the end of the frame before the next loop

**Comment by bpasero** [comment]
My theory is that possibly `closed` is actually `undefined` and so pushing a change to assert it is there.

https://github.com/microsoft/vscode/blob/71dcfe37ff048fe4f539aa6bb4cd4ada54679c16/extensions/vscode-api-tests/src/singlefolder-tests/workspace.test.ts#L135

**Comment by bpasero** [comment]
This failed again for web in: https://dev.azure.com/monacotools/Monaco/_build/results?buildId=182276&view=logs&j=a8b516b9-5c25-5ad9-505a-b03d38bc1dad&t=b2e05f10-ada9-503f-2369-d13399289d47&l=352

Given my change is in, it looks like literally even though we have `closed` document from the event, it is not the same as `doc`:

https://github.com/microsoft/vscode/blob/028186209f6b4db55b9cf10ab7edb26206ef237c/extensions/vscode-api-tests/src/singlefolder-tests/workspace.test.ts#L138-L149

I am pushing a change to compare the URIs of the two and see what they are.

**Comment by bpasero** [comment]
I see no failure since, let's close until we hit this again.

**Comment by bpasero** [comment]
Failed again in:

https://dev.azure.com/monacotools/Monaco/_build/results?buildId=184110&view=logs&j=a8b516b9-5c25-5ad9-505a-b03d38bc1dad&t=b2e05f10-ada9-503f-2369-d13399289d47&l=349

```
 AssertionError [ERR_ASSERTION] [ERR_ASSERTION]: closed.uri = fake-fs:/f7c6a90cfe9c7c87 but doc.uri = untitled:/__w/1/s/extensions/vscode-api-tests/testWorkspace/newfile.txt
```

This looks as if a document from a previous test run appears in our event listener. I guess we cannot guarantee that the event that comes in is for the closed editor, there maybe multiple. So an idea to fix this is to keep closed documents in an array for the test and try to find the one we expect.
